distsql/request_builder.go

```diff
@@ -227,11 +230,13 @@ func (builder *RequestBuilder) SetFromSessionVars(sv *variable.SessionVars) *Req
 	builder.Request.TaskID = sv.StmtCtx.TaskID
 	builder.Request.Priority = builder.getKVPriority(sv)
 	builder.Request.ReplicaRead = sv.GetReplicaRead()
+	builder.Request.TxnScope = sv.TxnScope
 	if sv.SnapshotInfoschema != nil {
 		builder.Request.SchemaVar = infoschema.GetInfoSchemaBySessionVars(sv).SchemaMetaVersion()
 	} else {
 		builder.Request.SchemaVar = sv.TxnCtx.SchemaVersion
 	}
+	builder.verifyTxnScope(sv)
```

Comment:
There's no strict order between `SetFromSessionVars` and `SetXXXRanges`. So move the check to `Build` should be much safer?

Reply:
nice catch, updated.